applications/dashboard/controllers/api/MediaApiController.php

```diff
@@ -292,15 +304,36 @@ public function normalizeOutput(array $row) {
     public function post(array $body) {
         $this->permission('Garden.Uploads.Add');
 
+        switch ($body['type']) {
+            case self::TYPE_TEXT:
+                $uploadExtensions = self::EXTENSIONS_TEXT;
+                break;
+            case self::TYPE_BINARY:
+                $uploadExtensions = self::EXTENSIONS_BINARY;
+                break;
+            case self::TYPE_ARCHIVE:
+                $uploadExtensions = self::EXTENSIONS_ARCHIVE;
+                break;
+            case self::TYPE_IMAGE:
+            default:
+                $uploadExtensions = array_keys(ImageResizer::getExtType());
+        }
+
         $uploadSchema = new UploadedFileSchema([
-            'allowedExtensions' => array_keys(ImageResizer::getExtType())
+            'allowedExtensions' => $uploadExtensions
```

Comment:
Vanilla has a `Garden.Upload.AllowedFileExtensions` config that should hold an array of all extensions a site allows for uploads. Now that we're adding more extensions to this endpoint, it might make sense to get an intersection of the two arrays to ensure only the configured extensions are allowed.

Reply:
Wouldn't you still want to perform an intersection of allowed extensions? If a site doesn't allow GIFs, and a client submits a request to this endpoint with a type of "image", wouldn't they still be able to upload a GIF?